### PR TITLE
Add Kconfig option to reduce the brightness of the LED ring

### PR DESCRIPTION
**Pull request by ntamas**
This PR adds a Kconfig option that allows the user to reduce the brightness of the LED ring by shifting all RGB values to the right by a given number of bits. It can be used for custom builds when the LED ring at full brightness and the motors together would draw too much current. Useful values for the option are most likely 0, 1 or 2, but I allow the full range of 8 bits for sake of completeness.

### CODE REVIEW

File: src/deck/drivers/src/ledring12.c [hunk 1]
```diff
@@ -48,6 +48,7 @@
 #include "debug.h"
 
 #define NBR_LEDS  CONFIG_DECK_LEDRING_NBR_LEDS
+#define DIMMER    CONFIG_DECK_LEDRING_DIMMER
```
[the remark this answers is not in the document]
Comment (ntamas) [reply src/deck/drivers/src/ledring12.c:51]: It's okay for me either way, I just did not want to change the `NBR_LEDS` part (it was the easiest to go from existing code from the Kbuild-based one without changing more than one line), and then I made the `DIMMER` part to behave the same way as `NBR_LEDS`. Let me know if you want me to just use the `CONFIG_...` defines directly and I'll do that.

[the remark this answers is not in the document]
Comment (tobbeanton) [reply src/deck/drivers/src/ledring12.c:51]: I will bring this up tomorrow and get back with how we would like it to look like.

[the remark this answers is not in the document]
Comment (tobbeanton) [reply src/deck/drivers/src/ledring12.c:51]: @ntamas Let's use the kconfig string CONFIG_DECK_LEDRING_DIMMER directly. Makes the code more clear.
Can you also replace the NBR_LEDS for CONFIG_DECK_LEDRING_NBR_LEDS? A search and replace should be sufficient and a few alignment that needs to be fixed.

[the remark this answers is not in the document]
Comment (ntamas) [reply src/deck/drivers/src/ledring12.c]: Done.

**Review by tobbeanton — APPROVED** [review]
Looks very nice!

**Pull request merged**